definitions/npm/ramda_v0.21.x/flow_v0.28.x-v0.30.x/ramda_v0.21.x.js

```diff
@@ -500,8 +500,8 @@ declare module ramda {
   declare function dissocPath<T>(key: Array<string>, val:T, ...args: Array<void>): (src: {[k:string]:T}) => {[k:string]:T};
   declare function dissocPath<T>(key: Array<string>, val:T, src: {[k:string]:T}): {[k:string]:T};
 
-  declare function evolve<V,R>(fn: NestedObject<(x:any) => R>, ...rest: Array<void>): (src: NestedObject<any>) => NestedObject<R>;
-  declare function evolve<V,R>(fn: NestedObject<(x:any) => R>, src: NestedObject<any>): NestedObject<R>;
+  declare function evolve<R: NestedObject<any>>(fn: NestedObject<UnarySameTypeFn<any>>, ...rest: Array<void>): (src: R) => R;
```

Comment:
These two lines are not accurate. In Flow, polymorphic types represent exact values, not exact types. This means that ```src``` & the return type would have to be referentially transparent, which they are not, since they are being modified by ```fn``` argument.